include/internal/property.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_PROPERTY_H
+# define HEADER_PROPERTY_H
+
+typedef struct ossl_impl_store_st OSSL_IMPL_STORE;
+
+/* Implementation store functions */
+OSSL_IMPL_STORE *ossl_impl_store_new(void);
+void ossl_impl_store_free(OSSL_IMPL_STORE *cache);
+int ossl_impl_store_init(void);
+void ossl_impl_store_cleanup(void);
+int ossl_impl_store_add(OSSL_IMPL_STORE *cache,
+                        int nid, const char *properties,
+                        void *implementation,
+                        void (*implementation_destruct)(void *));
+int ossl_impl_store_remove(OSSL_IMPL_STORE *cache,
+                           int nid, const void *implementation);
+int ossl_impl_store_fetch(OSSL_IMPL_STORE *cache,
+                          int nid, const char *properties,
+                          const void **result);
```

Comment:
Please make that non-void.  There's a good chance there will be refcounted objects...

Reply:
Oh wait, hold on, I might be wrong.  Deleting this if I can